src/index.d.ts

```diff
@@ -168,7 +176,13 @@ export function original<T>(value: T): T | void
 export function isDraft(value: any): boolean
 
 export class Immer {
-    constructor(config: {useProxies?: boolean; autoFreeze?: boolean})
+    constructor(config: {
+        useProxies?: boolean
+        autoFreeze?: boolean
+        onAssign?: <T>(state: ImmerState<T>, prop: keyof T, value: any) => void
```

Comment:
Not sure if `<T>` adds much value here, since the callback is called with different types of T and a single T propbably doesn't describe any object in the tree.  On the other hand it isn't in the way either.

Reply:
I don't remember why I added that, so I'll just remove it. 👍